lib/kitchen/figure_element.rb

```diff
@@ -20,7 +20,25 @@ def initialize(node:, document: nil)
     # @return [Element, nil]
     #
     def caption
-      first('figcaption')
+      first('figcaption', "div[data-type='description']")
+    end
+
+    # Returns true if the figure is a child of another figure
+    #
+    # @return [Boolean]
+    #
+
+    def is_subfigure?
+      parent.node.to_s.end_with?('</figure>')
+    end
+
+    # Increments the count of this element in all of this element's ancestors
+    # unless it's a subfigure
+
+    def count_as_descendant
+      @ancestors.each_pair do |type, ancestor|
+        @counts_in[type] = ancestor.increment_descendant_count(short_type) unless is_subfigure?
```

Comment:
cool solution! much better than my idea.

here's a tweak:  instead of copying this whole method from `ElementBase`, modify the method in `ElementBase` to call another method to see if should be counted.  

```ruby
# element_base.rb
def count_as_descendant
  @ancestors.each_pair do |type, ancestor|
    @counts_in[type] = ancestor.increment_descendant_count(short_type) if countable?
  end
end

def countable?
  true
end
```

With this approach, subclasses can opt out of counting by overriding `countable?` instead of the whole `each_pair` loop, e.g. `FigureElement` can have:

```ruby
def countable?
  !subfigure?
end
```

Also an optimization to the `ElementBase` approach is to put the check first, because there's no need to do all of the iterating if `countable?` is false:

```ruby
# element_base.rb
def count_as_descendant
  return unless countable?

  @ancestors.each_pair do |type, ancestor|
    @counts_in[type] = ancestor.increment_descendant_count(short_type)
  end
end
```

You'll need to add to the element_base_spec.rb to test that `countable?` is used in counting.